schain/schainwriter.go

```diff
@@ -57,13 +57,26 @@ func (w SChainWriter) Write(req *openrtb2.BidRequest, bidder string) {
 		sourceCopy := *req.Source
 		req.Source = &sourceCopy
 	}
-	schain := openrtb_ext.ExtRequestPrebidSChain{
-		SChain: *selectedSChain,
+
+	schain := openrtb_ext.ExtRequestPrebidSChain{}
+
+	if selectedSChain != nil {
+		schain.SChain = *selectedSChain
 	}
+
+	if hostSChainInfo != "" {
```

Comment:
For the scenario where the lone schain node is the host defined node, we set the schain `complete` field to `0`, which is the default value for that field when an instance of `openrtb_ext.ExtRequestPrebidSChain` is created. The `ver` field is set to `""` by default but I believe that is required as you can see here [OpenRTB Object: SupplyChain](https://github.com/InteractiveAdvertisingBureau/openrtb/blob/master/supplychainobject.md#openrtb-object-supplychain). The question is what should it be?

Reply:
After further research, it appears that `ver` should be set to `"1.0"` which is the latest and sole version of `sellers.json` which is linked to from [here](https://iabtechlab.com/sellers-json/). The change log on `sellers.json` shows 1.0.